source/crc64_sw.c

```diff
@@ -538,18 +540,580 @@ static uint64_t crc64nvme_table[8][256] = {
+        0x9a3575d081143fb6, 0x361453f59f62d697, 0xc276399abdf8edf5, 0x6e571fbfa38e04d4, // [7][0xac]
+        0x371106d61f772d7d, 0x9b3020f30101c45c, 0x6f524a9c239bff3e, 0xc3736cb93ded161f, // [7][0xb0]
+        0x87979e4267af88fa, 0x2bb6b86779d961db, 0xdfd4d2085b435ab9, 0x73f5f42d4535b398, // [7][0xb4]
+        0x3c8fa0a7bde1bf46, 0x90ae8682a3975667, 0x64cceced810d6d05, 0xc8edcac89f7b8424, // [7][0xb8]
+        0x8c093833c5391ac1, 0x20281e16db4ff3e0, 0xd44a7479f9d5c882, 0x786b525ce7a321a3, // [7][0xbc]
+        0x3e36722692930f0c, 0x921754038ce5e62d, 0x66753e6cae7fdd4f, 0xca541849b009346e, // [7][0xc0]
+        0x8eb0eab2ea4baa8b, 0x2291cc97f43d43aa, 0xd6f3a6f8d6a778c8, 0x7ad280ddc8d191e9, // [7][0xc4]
+        0x35a8d45730059d37, 0x9989f2722e737416, 0x6deb981d0ce94f74, 0xc1cabe38129fa655, // [7][0xc8]
+        0x852e4cc348dd38b0, 0x290f6ae656abd191, 0xdd6d00897431eaf3, 0x714c26ac6a4703d2, // [7][0xcc]
+        0x280a3fc5d6be2a7b, 0x842b19e0c8c8c35a, 0x7049738fea52f838, 0xdc6855aaf4241119, // [7][0xd0]
+        0x988ca751ae668ffc, 0x34ad8174b01066dd, 0xc0cfeb1b928a5dbf, 0x6ceecd3e8cfcb49e, // [7][0xd4]
+        0x239499b47428b840, 0x8fb5bf916a5e5161, 0x7bd7d5fe48c46a03, 0xd7f6f3db56b28322, // [7][0xd8]
+        0x931201200cf01dc7, 0x3f3327051286f4e6, 0xcb514d6a301ccf84, 0x67706b4f2e6a26a5, // [7][0xdc]
+        0x124ee8e01bc945e2, 0xbe6fcec505bfacc3, 0x4a0da4aa272597a1, 0xe62c828f39537e80, // [7][0xe0]
+        0xa2c870746311e065, 0x0ee956517d670944, 0xfa8b3c3e5ffd3226, 0x56aa1a1b418bdb07, // [7][0xe4]
+        0x19d04e91b95fd7d9, 0xb5f168b4a7293ef8, 0x419302db85b3059a, 0xedb224fe9bc5ecbb, // [7][0xe8]
+        0xa956d605c187725e, 0x0577f020dff19b7f, 0xf1159a4ffd6ba01d, 0x5d34bc6ae31d493c, // [7][0xec]
+        0x0472a5035fe46095, 0xa8538326419289b4, 0x5c31e9496308b2d6, 0xf010cf6c7d7e5bf7, // [7][0xf0]
+        0xb4f43d97273cc512, 0x18d51bb2394a2c33, 0xecb771dd1bd01751, 0x409657f805a6fe70, // [7][0xf4]
+        0x0fec0372fd72f2ae, 0xa3cd2557e3041b8f, 0x57af4f38c19e20ed, 0xfb8e691ddfe8c9cc, // [7][0xf8]
+        0xbf6a9be685aa5729, 0x134bbdc39bdcbe08, 0xe729d7acb946856a, 0x4b08f189a7306c4b  // [7][0xfc]
+    }};
+
+/* Calculate a CRC-64 eight bytes at a time on a big-endian architecture. */
+static inline uint64_t s_crc64_sw_be(const uint8_t *input, int length, uint64_t prev_crc64) {
+    uint64_t crc = ~aws_swap_bytes_if_needed_64(prev_crc64);
+    // Read byte by byte until we reach an 8 byte aligned address
+    while (length > 0 && ((intptr_t)input & 7)) {
+        crc = (crc << 8) ^ crc64nvme_table_be[0][((crc >> 56) ^ *input++)];
+        length--;
+    }
+
+    int remaining = length;
+    // Once we are aligned, read 8 bytes at a time
+    const uint64_t *current = (const uint64_t *)(const void *)input;
+    while (remaining >= 8) {
+        uint64_t c1 = *current++ ^ crc;
+        crc = crc64nvme_table_be[0][c1 & 0xff] ^ crc64nvme_table_be[1][(c1 >> 8) & 0xff] ^
```

Comment:
I'm not smart enough to check this in under a day ... but it looks like the other one, and the tests pass so probably good 👍